src/AutoMapper/ConfigurationStore.cs

```diff
@@ -364,6 +364,17 @@ public TypeMap FindTypeMapFor(TypePair typePair)
             return typeMap;
         }
 
+        private bool CoveredByObjectMap(TypePair typePair)
+        {
+            return GetMappers().FirstOrDefault(m => m.IsMatch(new ResolutionContext(null, null, typePair.SourceType, typePair.DestinationType, new MappingOperationOptions(), new MappingEngine(this)))) != null;
+        }

```

Comment:
The really ugly part is newing up the MappingEngine. A while ago we chatted about making object mappers only receive a TypePair. I'm gonna revisit that - perhaps there's something in the middle that makes more sense than having to new up a whole ResolutionContext just to determine if there's a match.